code/game/objects/effects/effect_system/smoke.dm

```diff
@@ -273,139 +278,99 @@
 /////////////////////////////////////////
 
 //Xeno acid smoke.
-/obj/effect/particle_effect/smoke/xeno_burn
-	time_to_live = 9
-	color = "#86B028" //Mostly green?
-	anchored = 1
+/obj/effect/particle_effect/smoke/xeno
+	lifetime = 6
 	spread_speed = 7
-	amount = 1 //Amount depends on Boiler upgrade!
+	var/strength = 1 // Effects scale with the emitter's bomb_strength upgrades.
+
+/obj/effect/particle_effect/smoke/xeno/copy_stats(obj/effect/particle_effect/smoke/xeno/parent)
+	strength = parent.strength
+	return ..()
+
+/obj/effect/particle_effect/smoke/xeno/smoke_mob(mob/living/carbon/C)
+	if(lifetime < 1 || !istype(C))
+		return FALSE
+	if(C.stat == DEAD)
+		return FALSE
+	if(isxeno(C) || (isyautja(C) && prob(75)))
```

Comment:
Not sure if we need to snowflake preds into new code

Reply:
I feel like retaining a bit of legacy here until we get to an overhaul or something going, alas.

Reply:
Tbh most snowflake predator things are slowly being removed (they are no longer in prefs, whitelist has been borked for a long time, they're only left in the admin menu at this point)

Reply:
fine then.